taca/illumina/utils.py

```diff
@@ -0,0 +1,242 @@
+""" Divers utilities for Illumina runs processing """
+
+import csv
+import glob
+import os
+import shutil
+import xml.etree.ElementTree as ET
+
+
+from bs4 import BeautifulSoup
+from itertools import izip_longest
+
+from taca.utils import parsers
+from taca.utils import filesystem
+from taca.utils.config import CONFIG
+
+finished_run_indicator = CONFIG.get('storage', {}).get('finished_run_indicator',
+                                                       'RTAComplete.txt')
+
+def last_index_read(directory):
+    """Parse the number of the highest index read from the RunInfo.xml
+    """
+    read_numbers = [int(read.get("Number", 0)) for read in parsers.get_read_configuration(directory) if read.get("IsIndexedRead", "") == "Y"]
```

Comment:
just continue reading the list comprehension :)

 `...int(read.get("Number", 0)) for read in parsers.get_read_configuration(directory)...`